src/lib/controllers/ui-controller.js

```diff
@@ -1303,9 +1284,11 @@ export default class UIController {
    * If no URLS are provided, will reset grammar data.
    * @param {Array} urls
    */
-  updateGrammar (urls = []) {
-    if (urls.length > 0) {
-      this.store.commit('app/setGrammarRes', urls[0])
+  updateGrammar (data) {
+    if (data && data.urls && data.urls.length > 0) {
+      this.api.app.grammarData[data.languageID] = data.urls[0]
```

Comment:
I am not sure if this works right in a scenario where the user switches languages -- e..g take the following scenario:

user looks up a greek word and clicks on a grammar term to open the grammar. the grammar url for greek gets set.

user then looks up a latin word and clicks on a grammar term to open the grammar. the grammar url for latin gets set.

Now the grammarData object has urls set for both greek and latin, with no indication which is the current one.

Reply:
I have missed the moment of clicking on the grammar attribute.
Will add it.

Reply:
Ok I have fixed it - now it updates URL well and also I have returned restorToGrammarIndex - because for now res-client doesn't define if url is index or not.